src/github/nag.rs

```diff
@@ -185,7 +130,147 @@ fn update_proposal_review_status(proposal_id: i32) -> DashResult<()> {
     Ok(())
 }
 
+fn update_poll_response_status(poll_id: i32) -> DashResult<()> {
+    let conn = &*DB_POOL.get()?;
+    // this is an updated comment from the bot itself
+
+    // parse out each "responded" status for each user, then update them
+
+    let survey: Poll = poll::table.find(poll_id).first(conn)?;
+
+    // don't update any statuses if the poll is closed
+    if survey.poll_closed {
+        return Ok(());
+    }
+
+    let comment: IssueComment = issuecomment::table
+        .find(survey.fk_bot_tracking_comment)
```

Comment:
Reading this code now, I'm wondering whether things like this shouldn't also be impl'd as small methods on the `Poll` type to more clearly separate the logic of a poll from its db semantics.

i.e.

```rs
let comment: IssueComment = self.tracking_comment()?;
```